api/lib/domain/services/certification-cpf-service.js

```diff
@@ -8,129 +8,135 @@ const CpfValidationStatus = {
 };
 
 class CpfBirthInformationValidation {
-  constructor({ message, code, status, birthCountry, birthINSEECode, birthPostalCode, birthCity }) {
-    this.message = message;
-    this.status = status;
+  constructor({ birthCountry, birthINSEECode, birthPostalCode, birthCity, errors = [] } = { errors: [] }) {
     this.birthCountry = birthCountry;
     this.birthINSEECode = birthINSEECode;
     this.birthPostalCode = birthPostalCode;
     this.birthCity = birthCity;
-    this.code = code;
+    this.errors = errors;
   }
 
-  static failure({ certificationCandidateError, data }) {
+  get status() {
+    return this.errors.length === 0 ? CpfValidationStatus.SUCCESS : CpfValidationStatus.FAILURE;
+  }
+
+  failure({ certificationCandidateError, data }) {
     const message = certificationCandidateError.getMessage(data);
-    return new CpfBirthInformationValidation({
-      message,
-      status: CpfValidationStatus.FAILURE,
-      code: certificationCandidateError.code,
-    });
+    this.errors.push({ message, code: certificationCandidateError.code });
+    return this;
```

Comment:
**Question (mobreview) :** 

Quel est l’interêt de retourner le this sachant qu'on a déjà l'objet ? (pour Antoine)